lib/parser/parser-base.ts

```diff
@@ -21,7 +24,8 @@ export class ParserPosition {
     return this.num >= 0 && this.num < this.lexerTokens.length;
   }
   public getRangeToEndLine() {
-    return this.lexerTokens[this.num]?.range.copyExtendEndOfLine();
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return this.lexerTokens[this.num]!.range.copyExtendEndOfLine();
```

Comment:
throw a parserError if out of range

Reply:
Based on which token? you need a range for that...

Reply:
The last line of the file because we are out of range maybe?